### PR TITLE
Add password change params

### PR DESCRIPTION
**Pull request by smartguest**
Companion PR for https://github.com/microsoft/azuredatastudio/pull/21295

Adds in password reset functionality.

### CODE REVIEW

File: src/azdata.proposed.d.ts [hunk 1]
```diff
@@ -397,9 +386,23 @@ declare module 'azdata' {
...
 
+	export interface PasswordChangeResult {
+		/**
+		 * Whether the password change was successful
+		 */
+		result: boolean;
+		/**
+		 * Error message if the password change was unsuccessful
+		 */
+		errorMessage?: string | undefined;
```
Comment (alanrenmsft) [inline src/azdata.proposed.d.ts:403]: having the `?` mark is the indicator nullable, so you don't have to add the `| undefined` type

File: src/azdata.proposed.d.ts [hunk 2]
```diff
@@ -397,9 +386,23 @@ declare module 'azdata' {
 	}
 
 	export interface ConnectionProvider extends DataProvider {
+		/**
+		 * Changes the password for a connection profile.
```
Comment (alanrenmsft) [inline src/azdata.proposed.d.ts:390]: please state that this is for the password expiration scenario to avoid confusion.

**Review by Charles-Gagnon — COMMENTED** [review]
See my comments in the ADS provider - I'm not sure this stuff should be in the ConnectionProvider API as it is currently since it isn't something that is applicable to non-MSSQL providers.

File: src/azdata.proposed.d.ts (hunk 2 above)
Comment (Charles-Gagnon) [reply src/azdata.proposed.d.ts]: Extensions can get providers directly using azdata.dataprotocol.getProvider - and thus call changePassword themselves. So I don't think we should have implementation-specific comments like that (but also see my other comments about whether we should have these public APIs at all)

**Review by cheenamalhotra — APPROVED** [review]

**Pull request merged**